lldb/tools/lldb-dap/src-ts/debug-adapter-factory.ts

```
@@ -137,53 +157,59 @@ export class LLDBDapDescriptorFactory
 
     const dbgOptions = {
       env: {
-        ...executable?.options?.env,
         ...configEnvironment,
         ...env,
       },
     };
-    const dbgArgs = executable?.args ?? [];
```

There might be some downstream users who are actually bundling an `lldb-dap` binary into their re-packaged distribution of this VS-Code extension.

At least the commit message from https://github.com/llvm/llvm-project/commit/b5d4332286154838557a8ab5c76b794e85d946b3 indicates, that there were some thoughts that downstream consumers might inject the lldb-dap path when using this extension as a library.

@JDevlieghere @walter-erquinigo are you aware if anyone is re-packaging this VS-Code extension together with an lldb-dap binary and might be relying on setting the lldb-dap path through the `package.json`?

---

I'm not but I'd be curious to find out too.

---

Long story short, I don't think anyone ever did that.

I intended to do that when I was working on the mojo extension, but I never really do that because I was waiting for the upstream code to have a feature the mojo extension didn't have in order to do the switch, but that didn't happen last year. This year I'm not working on that company anymore, so I don't think anyone will do that in the foreseable future. I know that Meta has their own debugger extension, which is very big, and at least they shouldn't be repackaging it. I don't know about other companies

---

But anyway, I'm fine with this change if we want less ways to set the args, which is good. However, you should throw an exception if `executable` is not `undefined`, which would be an unintended flow that might not work the way users expect.

---

ok, if you are fine with it, I am also fine with it

---

Alright, this took a bit of re-organizing since the `DebugAdapterDescriptorFactory` does not gracefully stop the debug session if it returns `undefined`. Instead, VS Code shows a modal with an error message that suggests that the extension may not have activated properly which is... unhelpful in this case.

I've moved the logic for starting the server and prompting the user for input to a new `DebugConfigurationProvider` which can stop the session gracefully and even open the `launch.json` for editing if something goes wrong.

In order to facilitate this, I had to add two new properties to the launch configuration that are used by the `DebugAdapterDescriptorFactory` to tell VS Code how to launch the debug adapter:

- `debugAdapterHostname` - the hostname for an existing lldb-dap server
- `debugAdapterPort `- the port for an existing lldb-dap server

This has the added bonus of allowing the user to start their own lldb-dap process in server mode if they really wanted to.

The `DebugAdapterDescriptorFactory` will also throw an error if the executable parameter is set.